EIPS/eip-7015.md

```diff
@@ -29,22 +29,29 @@ This EIP takes advantage of the fact that contract addresses can be precomputed
 
 **Signing Mechanism**
 
-Creator consent is given by signing an [EIP-712](./eip-712.md) compatible message; all signatures compliant with this EIP MUST include all fields defined. The struct signed is:
+Creator consent is given by signing an [EIP-712](./eip-712.md) compatible message; all signatures compliant with this EIP MUST include all fields defined. The struct signed can be any arbitrary data that defines how to create the token; it must hashed in an EIP-712 compatible format with a proper EIP-712 domain. 
+
+The following shows some examples of structs that could be encoded into `structHash` (defined below):
 
 ```solidity
+// defines that a token can be created with a specified id and metadata uri
 struct TokenCreation {
-	bytes32 structHash;
+  uint256 tokenId;
```

Comment:
Can we remove the `tokenId` the standard's attribution applies to all token-ids so this example might be a little misleading/confusing.

Reply:
done, made it more generic